mnelab/mainwindow.py

```diff
@@ -443,8 +444,15 @@ def open_file(self, f, text, ffilter="*"):
     def export_file(self, f, text, ffilter="*"):
         """Export to file."""
         fname = QFileDialog.getSaveFileName(self, text, filter=ffilter)[0]
+        sig = signature(f)
+
+        args={"fname": fname}
+        for i in sig.parameters.items():
+            if(i[0]=="ffilter"):
+                args[i]=ffilter
+
         if fname:
-            f(fname, ffilter)
+            f(*args.values())
```

Comment:
I think this could be shortened to:

```python
        if fname:
            try:
                f(fname, ffilter)
            except TypeError:
                f(fname)
```